src/validators/twitter-username-validator.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Yoast\WP\SEO\Validators;
+
+use Yoast\WP\SEO\Exceptions\Validation\Abstract_Validation_Exception;
+use Yoast\WP\SEO\Exceptions\Validation\Invalid_Twitter_Username_Exception;
+
+/**
+ * The Twitter username validator class.
+ */
+class Twitter_Username_Validator extends Text_Field_Validator {
+
+	// phpcs:disable Squiz.Commenting.FunctionCommentThrowTag.WrongNumber -- Reason: The parent validate can throw too.
+
+	/**
+	 * Holds the regex validator instance.
+	 *
+	 * @var Regex_Validator
+	 */
+	protected $regex_validator;
+
+	/**
+	 * Constructs a verification validator instance.
+	 *
+	 * @param Regex_Validator $regex_validator The regex validator.
+	 */
+	public function __construct( Regex_Validator $regex_validator ) {
+		$this->regex_validator = $regex_validator;
+	}
+
+	/**
+	 * Validates if a value is a Twitter username.
+	 *
+	 * @param mixed $value    The value to validate.
+	 * @param array $settings Optional settings.
+	 *
+	 * @throws \Yoast\WP\SEO\Exceptions\Validation\Invalid_Type_Exception When the type of the value is not a string.
+	 * @throws Invalid_Twitter_Username_Exception When the value is not a valid Twitter username.
+	 *
+	 * @return string A valid string.
+	 */
+	public function validate( $value, array $settings = null ) {
+		$string = parent::validate( $value );
+
+		$twitter_username = \ltrim( $string, '@' );
```

Comment:
In the original, the value is first trimmed and then validated.
I guess it would be nice to type check before trimming to prevent errors? But I'm not sure if this reverse can lead to anything.